### PR TITLE
Adding a coding style page for Core Database

### PR DESCRIPTION
**Pull request by mbautin**

### CODE REVIEW

File: docs/content/latest/contribute/core-database/build-from-src-centos.md [hunk 1]
```diff
@@ -53,7 +53,7 @@ Update packages on your system, install development tools and additional package
 ```sh
 sudo yum update
 sudo yum groupinstall -y 'Development Tools'
-sudo yum install -y ruby perl-Digest epel-release ccache git python2-pip python-devel python3 python3-pip python3-devel
+sudo yum install -y sudo ruby perl-Digest epel-release ccache git python2-pip python-devel python3 python3-pip python3-devel which
```
Comment (polarweasel) [inline docs/content/latest/contribute/core-database/build-from-src-centos.md:56]: So how are you running this command if you don't have sudo installed? 🙃

File: docs/content/latest/contribute/core-database/build-from-src-centos.md [hunk 2]
```diff
@@ -70,7 +70,7 @@ You could also symlink them into another directory that is on your `PATH`.
 
 {{< note title="Note" >}}
 
-We also use [Linuxbrew](https://github.com/linuxbrew/brew) to provide some of the third-party dependencies on CentOS.
+We also use [Linuxbrew](https://github.com/linuxbrew/brew) to provide some of the third-party dependencies on CentOS. Linuxbrew allows us to create a portable package
```
Comment (polarweasel) [inline docs/content/latest/contribute/core-database/build-from-src-centos.md:73]:
```suggestion
We also use [Linuxbrew](https://github.com/linuxbrew/brew) to provide some of the third-party dependencies on CentOS. Linuxbrew allows us to create a portable package.
```

File: docs/content/latest/contribute/core-database/clion-setup.md [hunk 3]
```diff
@@ -23,19 +23,19 @@ There are two options for build systems that you can use with YugabyteDB, `cmake
 ### Configure a CLion project for YugabyteDB
 
 1. Click **File > Open…** to open the project root directory.
```
Comment (polarweasel) [inline docs/content/latest/contribute/core-database/clion-setup.md:25]: If we're redoing this file anyway, let's do it as a proper markdown list: each entry starts with `1. `, put a blank line between steps, and indent any sub-paragraphs (like line 37) by four spaces (and a blank line before and after).

File: docs/content/latest/contribute/core-database/clion-setup.md [hunk 4]
```diff
@@ -23,19 +23,19 @@ There are two options for build systems that you can use with YugabyteDB, `cmake
...
+
     ![Clion ninja options](/images/contribute/clion-cmake-options-ninja.png)
 
-3. Use **“File / Reload CMake Project"** - it should start building third party dependencies for YugabyteDB. 
+3. Use **“File / Reload CMake Project"** - it should start building third party dependencies for YugabyteDB.
 Building a third party can take tens of minutes and then CLion will start updating symbols which also can take a while.
-4. Run from the command line inside project root (omit `YB_USE_NINJA=0` if you want to use ninja): 
+4. Run from the command line inside project root (omit `YB_USE_NINJA=0` if you want to use ninja):
 
 ```YB_USE_NINJA=0 ./yb_build.sh```
```
Comment (polarweasel) [inline docs/content/latest/contribute/core-database/clion-setup.md:40]:
````suggestion
    ```sh
    YB_USE_NINJA=0 ./yb_build.sh
    ```
````

File: docs/content/latest/contribute/core-database/coding-style.md [hunk 5]
```diff
@@ -0,0 +1,525 @@
...
+
+Use 100-character line length limit instead of 80-character.
+
+### Formatting function declarations and definitions
+
+Use one of the following formatting styles for function declarations and definitions.
+
+#### All arguments on one line
+
+```c++
```
Comment (polarweasel) [inline docs/content/latest/contribute/core-database/coding-style.md:47]: Tagging the language as `c++` won't get source-highlighted. This is visible in the build preview: https://deploy-preview-8837--infallible-bardeen-164bc9.netlify.app/latest/contribute/core-database/coding-style/#all-arguments-on-one-line

````suggestion
```cpp
````

**Review by polarweasel — COMMENTED** [review]
Didn't have time to finish the review this evening, but will continue tomorrow.

File: docs/content/latest/contribute/core-database/build-from-src-centos.md (hunk 1 above)
Comment (mbautin) [reply docs/content/latest/contribute/core-database/build-from-src-centos.md:56]: OK, removed.

File: docs/content/latest/contribute/core-database/build-from-src-centos.md (hunk 2 above)
Comment (mbautin) [reply docs/content/latest/contribute/core-database/build-from-src-centos.md:73]: Rewrote this section a bit.

File: docs/content/latest/contribute/core-database/coding-style.md (hunk 5 above)
Comment (mbautin) [reply docs/content/latest/contribute/core-database/coding-style.md]: Fixed.

File: docs/content/latest/contribute/core-database/clion-setup.md (hunk 4 above)
Comment (mbautin) [reply docs/content/latest/contribute/core-database/clion-setup.md]: Done.

File: docs/content/latest/contribute/core-database/clion-setup.md (hunk 3 above)
Comment (mbautin) [reply docs/content/latest/contribute/core-database/clion-setup.md:25]: Made these changes.

**Comment by polarweasel** [comment]
The coding-style page really feels like it could be better organized. The C++ guidelines in particular feel like they are fairly haphazardly ordered.

**Review by polarweasel — APPROVED** [review]
LGTM. Let's revisit periodically.

**Pull request merged**